### PR TITLE
Improve react query config & minor fix on permissions list

### PR DESCRIPTION
**Pull request by masskoder**
### Scope of changes

This adds :
- react query stale changes 
- fix default permissions on Generate API Modal component

Fixes SC-XXXX

### Type of change

- [ ] new feature
- [x] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [x] other (improvement)

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: web/beacon-app/src/application/config/react-query.tsx [hunk 1]
```diff
@@ -4,15 +4,15 @@ const queryClient = new QueryClient({
   // define default options for all queries
   defaultOptions: {
     queries: {
-      staleTime: 0,
+      staleTime: Infinity,
```
Comment (elysee15) [inline web/beacon-app/src/application/config/react-query.tsx:7]: I think here you are disabling the caching feature by default. May I know why ?

Comment (masskoder) [reply web/beacon-app/src/application/config/react-query.tsx:7]: No, it does not disable the caching feature. React query has 2 interesting config properties cacheTime and StateTime.
- The cacheTime indicates how long a response should be cached before it is discarded. as we know the Cache time is set to 5 minutes by default
- StaleTime indicates how long a response is considered fresh after it has been stale for a certain period.

Our request does not need to be fresh in our case. The InvalidQuery method is used on mutation to request fresh data for that particular endpoint that why i set stateTime to infinity

Comment (masskoder) [reply web/beacon-app/src/application/config/react-query.tsx:7]: A staleTime of 0 means that when a request is made, react query will serve the cached data first, then make a background request and populate the new data if it differs from the cached data.

File: web/beacon-app/src/features/apiKeys/components/GenerateAPIKeyModal.tsx [hunk 2]
```diff
@@ -68,6 +66,24 @@ function GenerateAPIKeyModal({
 
   const { values, setFieldValue } = formik;
 
+  // it turns out that the permissions are not being set correctly when the full access is selected by default
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/apiKeys/components/GenerateAPIKeyModal.tsx:69]:
```suggestion
  // Sets permissions correctly when full access is selected by default.
```

File: web/beacon-app/src/features/apiKeys/components/GenerateAPIKeyModal.tsx [hunk 3]
```diff
@@ -68,6 +66,24 @@ function GenerateAPIKeyModal({
...
+  // so this should fix that
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/apiKeys/components/GenerateAPIKeyModal.tsx:70]: If the suggestion above is accepted then this line should be deleted.

File: web/beacon-app/src/features/home/components/GenerateApiKeyStep.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ export default function GenerateApiKeyStep() {
   const alreadyHasKeys = apiKeys?.api_keys?.length > 0;
 
   const onOpenGenerateAPIKeyModal = () => {
-    if (alreadyHasKeys) return;
+    // if (alreadyHasKeys) return;
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/home/components/GenerateApiKeyStep.tsx:24]: Should this be removed or will a separate story need to be created for this?

Comment (masskoder) [reply web/beacon-app/src/features/home/components/GenerateApiKeyStep.tsx:24]: good catch, I should bring it back.

**Review by daniellemaxwell — APPROVED** [review]
Thank you for explaining the decision for setting `staleTime` to `Infinity` and for how invalidate query may be used to refresh data!

**Pull request merged**